bindings/matrix-sdk-ffi/src/timeline.rs

```diff
@@ -1288,8 +1286,38 @@ impl From<PollKind> for RumaPollKind {
     }
 }
 
-#[derive(Clone, uniffi::Record)]
+impl From<RumaPollKind> for PollKind {
+    fn from(value: RumaPollKind) -> Self {
+        match value {
+            RumaPollKind::Disclosed => Self::Disclosed,
+            RumaPollKind::Undisclosed => Self::Undisclosed,
+            _ => {
+                info!("Unknown poll kind, defaulting to undisclosed");
+                Self::Undisclosed
+            }
+        }
+    }
+}
+
+#[derive(uniffi::Record)]
 pub struct PollAnswer {
     pub id: String,
     pub text: String,
 }
+
+impl From<PollResult> for TimelineItemContentKind {
+    fn from(value: PollResult) -> Self {
+        TimelineItemContentKind::Poll {
+            question: value.question,
+            kind: PollKind::from(value.kind),
+            max_selections: value.max_selections,
+            answers: value
+                .answers
+                .iter()
+                .map(|i| PollAnswer { id: i.id.clone(), text: i.text.clone() })
```

Comment:
Since this function takes ownership of the `PollResult`, we can take ownership of the answers here via `.into_iter()`, to avoid having to clone:

```suggestion
                .into_iter()
                .map(|i| PollAnswer { id: i.id, text: i.text })
```